client/cli/go/cmds/device.go

```diff
@@ -340,3 +345,40 @@ var deviceRmTagsCommand = &cobra.Command{
 		return rmTagsCommand(cmd, heketi.DeviceSetTags)
 	},
 }
+
+var deviceListCommand = &cobra.Command{
+	Use:   "list",
+	Short: "List all devices on a node",
+	Long:  "List all devices on a node",
+	Example: `  $ heketi-cli device list \
```

Comment:
Good point. I was wondering whether we would make it possible to list all devices:
- on a node
- in a cluster
- in the whole topology?

Reply:
I understand, I wasn't sure what the scope of this was meant to be either. 

However, listing all of them in the topology sounds like an overlap with topology info. I'd consider enhancing the formatting of topology info than make this command too generic.

Reply:
Well, listing all devices just 1 line per device will still give a more condensed
output and not cover everything that topology info does, so I'd still see a 
certain justification for this.

But what level do you propose for the "device list" command?

Btw, even we would require providing a node, I still find it most natural
and idiomatic in this case to provide it with a `--node` switch, because
the id to provide is not the noun of the cli (device)...